app/src/androidTest/java/com/sdp/swiftwallet/presentation/signIn/RegisterActivityTest.java

```diff
@@ -27,29 +27,39 @@
 import com.sdp.cryptowalletapp.R;
 import com.sdp.swiftwallet.common.FirebaseUtil;
 
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import javax.inject.Inject;
 import org.bouncycastle.pqc.crypto.newhope.NHOtherInfoGenerator.PartyU;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.RuleChain;
 
+@HiltAndroidTest
 public class RegisterActivityTest {
 
-    @Rule
+    @Inject FirebaseAuth db;
+
+    // Add testing rules
     public ActivityScenarioRule<RegisterActivity> registerScenario = new ActivityScenarioRule<>(RegisterActivity.class);
 
+    public HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule
+    public final RuleChain rule =
+        RuleChain.outerRule(hiltRule).around(registerScenario);
```

Comment:
Not sure how that works, I would really appreciate if you could give us a course about hilt ?
Like maybe with a powerpoint.
And like a presentation video if possible.
Also is it possible to do this in english and swedish ?
Jokes aside let's talk about it later

Reply:
RuleChain is very complicated, its a composition of multiple rules in a certain order !!